playbooks/roles/web/templates/etc/varnish/default.vcl.j2

```diff
@@ -33,54 +33,110 @@ acl purge {
+          remove req.http.Accept-Encoding;
+      }
+  }
+
+  # Large static files should be piped, so they are delivered directly to the end-user without
+  # waiting for Varnish to fully read the file first.
+  if (req.url ~ "^[^?]*\.(7z|avi|bz2|flac|flv|gz|mka|mkv|mov|mp3|mp4|mpeg|mpg|ogg|ogm|opus|rar|tar|tgz|tbz|txz|wav|webm|xz|zip)(\?.*)?$") {
+      unset req.http.Cookie;
+      return (pipe);
+  }
+
+  # Remove all cookies for static files.
+  if (req.url ~ "^[^?]*\.(bmp|css|csv|doc|docx|eot|gif|ico|jpeg|jpg|js|odt|otf|pdf|png|ppt|pptx|rtf|svg|svgz|swf|ttf|txt|webp|woff|woff2|xls|xlsx|xml)(\?.*)?$") {
+      unset req.http.Cookie;
+      return (lookup);
+  }
+
 {% if conf_varnish_vcl_recv is defined %}
-    {{ conf_varnish_vcl_recv }}
+  {{ conf_varnish_vcl_recv }}
 {% endif %}
+
+  # Don't cache requests with authorization. (http://book.varnish-software.com/3.0/VCL_Basics.html#default-vcl-recv)
+  if (req.http.Authorization) {
+      return (pass);
+  }
+
+  return (lookup);
 }
 
 #################
 # sub vcl_fetch #
 #################
 sub vcl_fetch {
-  # Many requests contain Accept-Encoding HTTP headers.
-  # We standardize and remove these when unnecessary to make it easier to cache requests.
-  if (req.http.Accept-Encoding) {
-    # If the request URL has any of these extensions, remove the Accept-Encoding header as it is meaningless.
-   if (req.url ~ "\.(jpe?g|jpe|gif|png|webp)$") {
-      remove req.http.Accept-Encoding;
-    # If the Accept-Encoding contains 'gzip' standardize it.
-    } elsif (req.http.Accept-Encoding ~ "gzip") {
-      set req.http.Accept-Encoding = "gzip";
-    # If the Accept-Encoding contains 'deflate' standardize it.
-    } elsif (req.http.Accept-Encoding ~ "deflate") {
-      set req.http.Accept-Encoding = "deflate";
-    # If the Accept-Encoding header isn't matched above, remove it.
-    } else {
-      remove req.http.Accept-Encoding;
-    }
-  }
-
-  # Strip any cookies before an image/JS/CSS is inserted into cache.
-  if (req.url ~ "\.(jpe?g|jpe|gif|png|webp|svg|css|js)$") {
+  # Strip any cookies before a static file is inserted into cache.
+  if (req.url ~ "^[^?]*\.(bmp|css|csv|doc|docx|eot|gif|ico|jpeg|jpg|js|odt|otf|pdf|png|ppt|pptx|rtf|svg|svgz|swf|ttf|txt|webp|woff|woff2|xls|xlsx|xml)(\?.*)?$") {
      unset beresp.http.set-cookie;
   }
 
-  #set obj.grace = 5m;
-  set beresp.grace = 2m;
+  # Allow stale content, in case the backend goes down. (Old content to show is better than no content at all.)
+  set beresp.grace = 6h;
 
-{% if conf_varnish_vcl_fetch is defined %}
-  {{ conf_varnish_vcl_fetch }}
-{% endif %}
+  {% if conf_varnish_vcl_fetch is defined %}
+    {{ conf_varnish_vcl_fetch }}
+  {% endif %}
+
+  # Set 2min cache if unset for static files. (http://book.varnish-software.com/3.0/VCL_Basics.html#default-vcl-fetch)
+  if (beresp.ttl <= 0s || beresp.http.Set-Cookie || beresp.http.Vary == "*") {
+      set beresp.ttl = 120s;
+      return (hit_for_pass);
+  }
```

Comment:
Copied from the [default config](http://book.varnish-software.com/3.0/VCL_Basics.html#default-vcl-fetch).